src/main/java/jewellery/inventory/service/UserInOrganizationService.java

```diff
@@ -0,0 +1,191 @@
+import jewellery.inventory.exception.organization.MissingOrganizationPermissionException;
+import jewellery.inventory.exception.organization.UserIsNotPartOfOrganizationException;
+import jewellery.inventory.mapper.OrganizationMapper;
+import jewellery.inventory.model.*;
+import jewellery.inventory.repository.OrganizationRepository;
+import jewellery.inventory.repository.UserInOrganizationRepository;
+import jewellery.inventory.service.security.AuthService;
+import lombok.AllArgsConstructor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class UserInOrganizationService implements EntityFetcher {
+  private static final Logger logger = LogManager.getLogger(UserInOrganizationService.class);
+  private final OrganizationRepository organizationRepository;
+  private final OrganizationMapper organizationMapper;
+  private final AuthService authService;
+  private final UserService userService;
+  private final UserInOrganizationRepository userInOrganizationRepository;
+  private final OrganizationService organizationService;
+
+  public OrganizationMembersResponseDto getAllUsersInOrganization(UUID organizationId) {
+    Organization organization = organizationService.getOrganization(organizationId);
+    validateUserInOrganization(organization);
+    return organizationMapper.toOrganizationMembersResponseDto(organization);
+  }
+
+  @LogUpdateEvent(eventType = EventType.ORGANIZATION_USER_UPDATE)
+  public OrganizationSingleMemberResponseDto updateUserPermissionsInOrganization(
+      UUID userId, UUID organizationId, List<OrganizationPermission> organizationPermissionList) {
+
+    UserInOrganization userInOrganization =
+        userInOrganizationRepository
+            .findByUserIdAndOrganizationId(userId, organizationId)
+            .orElseThrow(() -> new UserIsNotPartOfOrganizationException(userId, organizationId));
+
+    Organization organization = userInOrganization.getOrganization();
+    User userForUpdate = userInOrganization.getUser();
+
+    validateCurrentUserPermission(organization, OrganizationPermission.MANAGE_USERS);
+    changeUserPermissionInOrganization(organization, userForUpdate, organizationPermissionList);
+    logger.info(
+        "Successfully updated user permissions in the organization. Organization ID: {}, User ID: {}",
+        organizationId,
+        userId);
+    return organizationMapper.toOrganizationSingleMemberResponseDto(userId, organization);
+  }
+
+  @LogCreateEvent(eventType = EventType.ORGANIZATION_USER_CREATE)
+  public OrganizationSingleMemberResponseDto addUserInOrganization(
+      UUID organizationId, UserInOrganizationRequestDto userInOrganizationRequestDto) {
+    Organization organization = organizationService.getOrganization(organizationId);
+    validateCurrentUserPermission(organization, OrganizationPermission.MANAGE_USERS);
```

Comment:
same here with the validation maybe
```suggestion
    validateCurrentUserPermission(organization, OrganizationPermission.MANAGE_USERS);
    Organization organization = organizationService.getOrganization(organizationId);
```